pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -352,9 +352,18 @@ func (r *RootSyncReconciler) SetupWithManager(mgr controllerruntime.Manager, wat
 			MaxConcurrentReconciles: 1,
 		}).
 		For(&v1beta1.RootSync{}).
-		Owns(&appsv1.Deployment{}).
-		Watches(&source.Kind{Type: &corev1.Secret{}},
+		// Custom Watch to trigger Reconcile for objects created by RootSync controller.
```

Comment:
> Add watches for all objects managed by the reconciler-manager for the RootSync. This should allow the reconciler-manager to revert any drift on the objects it manages, at least on fields it specifies.

Could you add some unit and/or e2e tests for this functionality?

Reply:
I was afraid you were going to ask this. There's no unit tests that actually spin up these controllers with the controller manager. So I'm going to have to scaffold that up first and solve any problems when trying to run it with the fake client.

Reply:
Added with https://github.com/GoogleContainerTools/kpt-config-sync/pull/634